source/manuals/programming-languages/html.html.md.erb

```diff
@@ -94,6 +69,31 @@ Examples include author information, related links and related content.
 
 `<footer role="contentinfo">` should be used for the footer of the site.
 
+### Region landmarks
+
+Region landmarks can be used when you need to create a new type of landmark.
+
+You can create region landmarks by using the `<section>` tag with an
+`aria-labelledby` attribute linking to the id of the heading that identifies
+it.
+
+Region landmarks should only be used where:
+
+- users will likely want to be able to navigate to the grouping easily and to
+  have it listed in a summary of the page
+- none of the existing HTML5 tags or ARIA landmark roles used for document
+  structure would correctly identify the grouping
+
+The use of region landmarks should be limited, as too many landmarks can dilute
+their usefulness in quickly navigating to important areas of a page. If you
+find there are many custom landmarks then the page’s structure should be
+re-examined.
```

Comment:
I think this is a direct quote from https://www.scottohara.me/blog/2018/03/03/landmarks.html, so suggest we credit it as such and/or work the relevant points into our own guidance.

Would also be tempted to put into active voice and provide more of a steer on what we mean when we say use should be 'limited'. Maybe something like:

> Only use a region landmark if [you have evidence it's useful for your users?]. Adding too many landmarks to a page makes it harder for users to quickly understand the structure of the page or to use them to navigate.

I've also removed the last sentence as '_many_ custom landmarks' is too subjective IMO. When do you need to re-examine the page structure? When you have more than 3 custom landmarks? More than 100? 🤷🏻 Maybe we could add something about doing user research with disabled users / users who use landmarks to navigate instead?

Reply:
It is a bit vague. I think what's being hinted at here is that you only really want to start wrapping content in labelled regions when you've exhausted other methods so doing it should be really rare. In this order, I think you should be approaching it something like this:
1. sectioning your content by labelling it with headings
2. identifying which of those sections persist between pages, include things you'd want to access regularly or are a grouping that might have different headings but users think of as one thing and find the right landmark for them
3. use a region landmark if there isn't another landmark that works and naming it something that describes the semantics of the section
4. see if this helps with tasks more than the weight it adds to the mental model and how hard it is to use the other landmarks, for you and real users

In other words, using a region landmark should be the last sectioning tool you use. I also think this is kind of what's implied by the idea that using a region landmarks can mean you've jumped ahead and aren't considering headings as a useful enough sectioning tool.

Phew! Be interested in your thoughts. If what I described here makes some sense, could this warning be written a bit more like: 'you can already use headings and the other landmarks to identify sections so region landmarks should really be used when those are proven not to work well enough and doing so saves users more effort than it adds'?

Reply:
Makes sense to me, and your suggestion sounds good too 👍🏻